src/HtmlProcessor/CssToAttributeConverter.php

```diff
@@ -64,19 +69,25 @@ public function convertCssToVisualAttributes(): self
 
     /**
      * Returns a list with all DOM nodes that have a style attribute.
+     *
+     * @return \DOMNodeList<\DOMElement>
      */
     private function getAllNodesWithStyleAttribute(): \DOMNodeList
     {
-        return $this->getXPath()->query('//*[@style]');
+        $result = $this->getXPath()->query('//*[@style]');
+        \assert($result instanceof \DOMNodeList);
+        /** @var \DOMNodeList<\DOMElement> $result */
+
+        return $result;
     }
 
     /**
      * Applies `$styles` to `$node`.
      *
-     * This method maps CSS styles to HTML attributes and adds those to the
-     * node.
+     * This method maps CSS styles to HTML attributes and adds those to the node.
      *
-     * @param array<string, string> $styles the new CSS styles taken from the global styles to be applied to this node
+     * @param array<non-empty-string, string> $styles the new CSS styles taken from the global styles
+     *        to be applied to this node
```

Comment:
The description could be all on its own separate line, for readability:

```suggestion
     * @param array<non-empty-string, string> $styles
     *        the new CSS styles taken from the global styles to be applied to this node
```